Firebase/Auth/Source/FIRAuth.m

```diff
@@ -1444,19 +1443,47 @@ - (nullable NSData *)APNSToken {
   return result;
 }
 
-- (void)setAPNSToken:(nullable NSData *)APNSToken {
-  [self setAPNSToken:APNSToken type:FIRAuthAPNSTokenTypeUnknown];
+#pragma mark - UIApplicationDelegate
+
+- (void)application:(UIApplication *)application
+didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken {
+  [self setAPNSToken:deviceToken type:FIRAuthAPNSTokenTypeUnknown];
 }
 
-- (void)setAPNSToken:(NSData *)token type:(FIRAuthAPNSTokenType)type {
+- (void)application:(UIApplication *)application
+didFailToRegisterForRemoteNotificationsWithError:(NSError *)error {
   dispatch_sync(FIRAuthGlobalWorkQueue(), ^{
-    self->_tokenManager.token = [[FIRAuthAPNSToken alloc] initWithData:token type:type];
+    [self->_tokenManager cancelWithError:error];
   });
 }
 
-- (void)handleAPNSTokenError:(NSError *)error {
+- (void)application:(UIApplication *)application
+didReceiveRemoteNotification:(NSDictionary *)userInfo
+fetchCompletionHandler:(void (^)(UIBackgroundFetchResult))completionHandler {
+  [self canHandleNotification:userInfo];
+}
```

Comment:
If Auth can handle the notification, we want to stop passing the notification to other UIApplicationDelegate interceptors. Is it possible with GULAppDelegateSwizzler?

Reply:
`GULAppDelegateSwizzler` is a generic class that makes sure the methods are called for each interceptor as well as for the original implementation. AFAIK it would be a violation of the class design if one interceptor could affect other interceptors (e.g. prevent calling a method). @tejasd @ryanwilson please let us know if it is not true.

@renkelvin If it is an essential to call the methods on only one arbitrary instance of `FIRAuth`  I can recover `FIRAuthAppDelegateProxy` and put this logic there. What do you think?

Reply:
GULAppDelegateSwizzler cannot currently handle this scenario.

@renkelvin Would you be able to elaborate on why that's necessary? At first glance, that seems like a dangerous thing to do.

Reply:
@tejasd Yeah, I agree it seems dangerous. Auth uses remote notification and redirected URL to pass some sensitive information. These information are necessary but not sufficient to hack a user's account. It would be better if we can hide them if possible.

@maksymmalyhin Yeah, it's not what `GULAppDelegateSwizzler` should handle. I agree.

Let me double check with my team to see if we can find a better solution and get back to you guys. Thanks!

Reply:
Hi all, after further discussions, we think Auth should follow the pattern of `GULAppDelegateSwizzler` and we'll be more restricted to the information passed in notifications and urls.
@maksymmalyhin `FIRAuthAppDelegateProxy` can't fully protect the sensitive information, so let's still discard it. Thank you!